### DISCUSSION TITLE
How to add multiple keystrokes to a TreeView node?

### DISCUSSION THREAD

**Discussion by Andras-Csanyi** [tui-cs/Terminal.Gui#3589] [Q&A] [answered]
I'd like to add multiple keystrokes to a TreeView node achieve something like the following:

- the focus is on an item
- I hit enter a dialog opens up and I see the details of the item (this can be done by the `ObjectAcivated`)
- the focus is still on the item in a node and I hit `Shift-r` and a second operation can be executed where the details of the node are needed
- the focus is still on the item in a node and I hit `Shit-a` and a third operation can be executed where the details of the node are needed

I am reading the API of the TreeView and View and it seems this is not possible, because only the `ObjectActivated` method gets the node.

Thanks!

**Comment by tznind** [comment, marked as answer]
Here is an example of using the `KeyPress` event to perform different actions in a `TreeView<FileSystemInfo>`.

```csharp
            // Prevent selecting many objects at once
            tv.MultiSelect = false;
            // Disable normal activation key
            tv.ObjectActivationKey = Key.Null;

            tv.KeyPress += (e) =>
            {

                // Use 'as' your class here or do conditional checks
                var o = tv.SelectedObject as FileSystemInfo;

                if (o == null)
                {
                    return;
                }

                if (e.KeyEvent.Key == Key.A)
                {
                    e.Handled = true;
                    MessageBox.Query("Last Write", o.LastAccessTime.ToString(), "Ok");
                }
                if (e.KeyEvent.Key == (Key.Enter))
                {
                    e.Handled = true;
                    MessageBox.Query("Full Path", o.ToString(), "Ok");
                }
                if (e.KeyEvent.Key == Key.R && o is FileInfo f)
                {
                    e.Handled = true;
                    MessageBox.Query("Length", f.Length + " bytes", "Ok");
                }
            };
```

**Reply by Andras-Csanyi** [reply]
Thank you very much! Is there an example for this among the scenarios? If not I might create one.

**Reply by tznind** [reply]
No problem! 

> Is there an example for this among the scenarios? If not I might create one.

Not really for tree view, closest is [here in the TreeViewFileSystem scenario](https://github.com/gui-cs/Terminal.Gui/blame/1816529ca2d03764dd3ad93d8166b752798db43f/UICatalog/Scenarios/TreeViewFileSystem.cs#L452) but it only opens a context menu for an action that happens automatically on object selected anyway 🤦 


Currently v1 is in maintenence only mode and development work is on the v2_develop branch (nuget packages are available for v2 prerelease too btw).

If you wanted to add more key handling to this scenario in v2 that would be most welcome!
